index.js

```diff
@@ -25,14 +27,30 @@ module.exports = {
       name: 'ember-cli-babel',
       ext: 'js',
       toTree: function(tree) {
-        return require('broccoli-babel-transpiler')(tree, getBabelOptions(addon));
+        return require('broccoli-babel-transpiler')(tree, addon._getBabelOptions());
       }
     });
   },
 
   shouldIncludePolyfill: function() {
-    var options = getAddonOptions(this);
-    return options.includePolyfill === true;
+    var addonOptions = this._getAddonOptions();
+    var babelOptions = addonOptions.babel;
+    var customOptions = addonOptions['ember-cli-babel'];
+
+    if (customOptions && 'includePolyfill' in customOptions) {
+      return customOptions.includePolyfill === true;
```

Comment:
@rwjblue since the deprecation will only be triggered in recent CLIs we can just use `writeDeprecateLine` here

@nathanhammond I'll move the deprecation check in front of everything